File: jung-visualization/src/test/java/edu/uci/ics/jung/visualization/BasicVisualizationServerTest.java
package edu.uci.ics.jung.visualization;

import com.google.common.graph.Network;
import com.google.common.graph.NetworkBuilder;
import edu.uci.ics.jung.layout.algorithms.CircleLayoutAlgorithm;
import edu.uci.ics.jung.visualization.picking.PickedState;
import java.awt.*;
import junit.framework.TestCase;

public class BasicVisualizationServerTest extends TestCase {

  /*
   * Previously, a bug was introduced where the RenderContext in BasicVisualizationServer was reassigned, resulting
   * in data like pickedNodeState to be lost.
   */
  public void testRenderContextNotOverridden() {
    Network<Object, Object> graph = NetworkBuilder.directed().build();
    CircleLayoutAlgorithm algorithm = new CircleLayoutAlgorithm();

    BasicVisualizationServer server =
        new BasicVisualizationServer<Object, Object>(graph, algorithm, new Dimension(600, 600));

    PickedState<Object> pickedNodeState = server.getRenderContext().getPickedNodeState();
    assertNotNull(pickedNodeState);
  }
}

File: jung-visualization/src/test/kotlin/edu/uci/ics/jung/visualization/BasicVisualizationServerTest.kt
package edu.uci.ics.jung.visualization

import com.google.common.graph.NetworkBuilder
import edu.uci.ics.jung.layout.algorithms.CircleLayoutAlgorithm
import java.awt.Dimension
import junit.framework.TestCase

class BasicVisualizationServerTest : TestCase() {

  /*
   * Previously, a bug was introduced where the RenderContext in BasicVisualizationServer was reassigned, resulting
   * in data like pickedNodeState to be lost.
   */
  fun testRenderContextNotOverridden() {
    val graph = NetworkBuilder.directed().build<Any, Any>()
    val algorithm = CircleLayoutAlgorithm<Any>()

    val server = BasicVisualizationServer<Any, Any>(graph, algorithm, Dimension(600, 600))

    val pickedNodeState = server.getRenderContext().getPickedNodeState()
    assertNotNull(pickedNodeState)
  }
}
